spec/03-plugins/19-acl/01-api_spec.lua

```diff
@@ -268,4 +269,153 @@ describe("Plugin: acl (API)", function()
       end)
     end)
   end)
+  
+  describe("/acls", function()
+    local consumer2
+
+    describe("GET", function()
+      setup(function()
+        helpers.dao:truncate_table("acls")
+
+        for i = 1, 3 do
+          assert(helpers.dao.acls:insert {
+            group = "group" .. i,
+            consumer_id = consumer.id
+          })
+        end
+
+        consumer2 = assert(helpers.dao.consumers:insert {
+          username = "bob-the-buidler"
+        })
+
+        for i = 1, 3 do
+          assert(helpers.dao.acls:insert {
+            group = "group" .. i,
+            consumer_id = consumer2.id
+          })
+        end
+      end)
+
+      it("retrieves all the acls with trailing slash", function()
+        local res = assert(admin_client:send {
+          method = "GET",
+          path = "/acls/",
+        })
+        local body = assert.res_status(200, res)
+        local json = cjson.decode(body)
+        assert.is_table(json.data)
+        assert.equal(6, #json.data)
+        assert.equal(6, json.total)
+      end)
+      it("retrieves all the acls without trailing slash", function()
+        local res = assert(admin_client:send {
+          method = "GET",
+          path = "/acls",
+        })
+        local body = assert.res_status(200, res)
+        local json = cjson.decode(body)
+        assert.is_table(json.data)
+        assert.equal(6, #json.data)
+        assert.equal(6, json.total)
+      end)
+      it("paginates through the acls", function()
+        local res = assert(admin_client:send {
+          method = "GET",
+          path = "/acls?size=3",
+        })
+        local body = assert.res_status(200, res)
+        local json_1 = cjson.decode(body)
+        assert.is_table(json_1.data)
+        assert.equal(3, #json_1.data)
+        assert.equal(6, json_1.total)
+
+        res = assert(admin_client:send {
+          method = "GET",
+          path = "/acls?size=3&offset=" .. json_1.offset,
+        })
+        body = assert.res_status(200, res)
+        local json_2 = cjson.decode(body)
+        assert.is_table(json_2.data)
+        assert.equal(3, #json_2.data)
+        assert.equal(6, json_2.total)
+
+        assert.not_same(json_1.data, json_2.data)
+        assert.is_nil(json_2.offset) -- last page
+      end)
```

Comment:
I had a doubt and tried to run this test with Cassandra, and encountered a few errors that made it fail. Here is the patch I had to add to make this test succeed with Cassandra: 

```diff
diff --git a/spec/03-plugins/19-acl/01-api_spec.lua b/spec/03-plugins/19-acl/01-api_spec.lua
index 99ad25f9..31435e59 100644
--- a/spec/03-plugins/19-acl/01-api_spec.lua
+++ b/spec/03-plugins/19-acl/01-api_spec.lua
@@ -269,7 +269,7 @@ describe("Plugin: acl (API)", function()
       end)
     end)
   end)
-  
+
   describe("/acls", function()
     local consumer2
 
@@ -331,7 +331,11 @@ describe("Plugin: acl (API)", function()
 
         res = assert(admin_client:send {
           method = "GET",
-          path = "/acls?size=3&offset=" .. json_1.offset,
+          path = "/acls",
+          query = {
+            size = 3,
+            offset = json_1.offset,
+          }
         })
         body = assert.res_status(200, res)
         local json_2 = cjson.decode(body)
@@ -340,7 +344,10 @@ describe("Plugin: acl (API)", function()
         assert.equal(6, json_2.total)
 
         assert.not_same(json_1.data, json_2.data)
-        assert.is_nil(json_2.offset) -- last page
+        -- Disabled: on Cassandra, the last page still returns a
+        -- next_page token, and thus, an offset proprty in the
+        -- response of the Admin API.
+        --assert.is_nil(json_2.offset) -- last page
       end)
       it("retrieves acls for a consumer_id", function()
         local res = assert(admin_client:send {
```

It is likely the case for other contributions we've recently commited that embed the same test.